lua/autorun/server/bounties.lua

```diff
@@ -0,0 +1,72 @@
+
+hook.Add("PlayerDeath","BountyDeath",function(ply,infl,atk)
+	local bounty = GetPlayerBounty(ply)
+	if bounty > 0 and ply != atk and type(atk) == "Player" then
+		SetPlayerBounty(ply,0)
+		atk:PS_GivePoints(bounty)
+		BotSayGlobal("[edgy]"..atk:Nick().." [fbc]has claimed [gold]"..ply:Nick().."'s [fbc]bounty of [rainbow]"..bounty.." [fbc]points!")
+	end
+end)
+
+function GetPlayerBounty(ply)
+	if ply.bounty == nil then
+		ply.bounty = tonumber(ply:GetPData("bounty",0))
+	end
+	return ply.bounty
+end
+
+function SetPlayerBounty(ply,bounty)
+	ply:SetPData("bounty",bounty)
+	ply.bounty = bounty
+end
+
+function AddBounty(ply,target,amount)
+	if type(ply) != "Player" or type(target) != "Player" or type(amount) != "number" then return end
+	amount = amount > 0 and amount or 0
+	if ply:PS_HasPoints(amount) then
+		SetPlayerBounty(target,GetPlayerBounty(target) + amount)
+		ply:PS_TakePoints(amount)
+		BotSayGlobal("[fbc]"..target:Nick().."'s bounty is now [rainbow]"..GetPlayerBounty(target) + amount.." [fbc]points")
+	else
+		ply:ChatPrint("[red]You don't have enough points")
+	end
+end
+
+RegisterChatCommand({'bounty'},function(ply,arg)
+	local t = string.Explode(" ",arg)
+	local to = PlyCount(t[1])
+	if type(to[1]) == "Player" and not to[2]>1 then
+		local bounty = GetPlayerBounty(to)
+		local p = tonumber(t[2])
```

Comment:
An issue here is, what if someone's username is, for example, "trump 6000000", and someone types !bounty trump 6000000 1000 intending to put a 1000 point bounty. A decent fix here would be to use t[#t] instead. Although I think you should actually make two commands, !bounty to see what the bounty is and !setbounty to actually charge points

Reply:
I think sticking to just !bounty is better because it's more intuitive and isn't this how it is for !givepoints?

Reply:
It's not the same because !givepoints only has one function. This one both checks the current bounty AND adds points to the bounty. I think it should be !setbounty so it is explicit that you're trying to set the points (from experience i am quite sure that with this someone would accidentally set a huge bounty sooner or later and demand their points back...)

Reply:
btw, if you just do amount = table.remove(t) and then string.Implode t with a space, you'll get the whole argument back except for the last number. That way if someone's name is "T R U M P" it won't just search for T and fail

Reply:
how about something like !showbounty, I'd like to keep the addition to bounties tied to !bounty

Reply:
Sure